src/elements/toniq-slider/toniq-slider.element.ts

```diff
@@ -0,0 +1,407 @@
+        }
+
+        .slider-wrapper {
+            position: relative;
+            width: 100%;
+        }
+
+        .slider::-webkit-slider-thumb {
+            -webkit-appearance: none;
+            position: relative;
+            height: 16px;
+            width: 16px;
+            z-index: 10;
+            cursor: pointer;
+            pointer-events: all;
+            border-radius: 10px;
+            transition: ${interactionDuration};
+            ${applyBackgroundAndForeground(toniqColors.accentPrimary)};
+        }
+
+        .slider::-webkit-slider-thumb:hover {
+            transform: scale(1.2);
+        }
+    `,
+    initCallback: ({props, setProps}) => {
+        setProps({value: props.value < props.min ? props.min : props.value});
+    },
+    renderCallback: ({props, host, dispatch, events}) => {
+        function onSliderResize(
+            entry: Readonly<Pick<ResizeObserverEntry, 'target' | 'contentRect'>>,
+        ) {
+            if (props.double) {
+                fillDoubleRangeSlider();
+            } else {
+                fillRangeSlider(parseInt((entry.target as HTMLInputElement).value));
+            }
+        }
+
+        function fillRangeSlider(value: number) {
+            const slider = host.shadowRoot?.querySelector('.slider') as HTMLInputElement;
+            const progress = host.shadowRoot?.querySelector('.progress') as HTMLElement;
+            const label = host.shadowRoot?.querySelector('.label') as HTMLElement;
+            const sliderWidth = slider.clientWidth;
+
+            const sliderOffset = (8 * value) / props.max;
+            const progressOffset = ((value - props.min) * (0 - 8)) / (props.max - props.min) + 8;
+
+            const progressWidth =
+                (sliderWidth * valueMap(value, props.min, props.max, 0, props.max)) / props.max -
+                sliderOffset;
+            progress.style.width = `${progressWidth + progressOffset}px`;
+
+            label.innerHTML = `${slider.value} ${props.suffix}`;
+            /**
+             * Since the input range thumb is a pseudo element, the trick to getting its location is
+             * by getting the progress bar right location minus half of the label value width. Then
+             * set label value left to that computed value.
+             */
+            const labelOffset = progress.getBoundingClientRect().right - label.offsetWidth / 2;
+
+            label.style.left = `${
+                labelOffset < slider.getBoundingClientRect().left
+                    ? slider.getBoundingClientRect().left
+                    : labelOffset + label.clientWidth > slider.getBoundingClientRect().right
+                    ? slider.getBoundingClientRect().right - label.clientWidth
+                    : labelOffset
+            }px`;
+
+            // Plus 16px (8px half of font size + 8px spacing from thumb to label based on design)
+            label.style.top = `${progress.getBoundingClientRect().top + 16}px`;
```

Comment:
Same idea, don't imperatively set this here, set it in the HTML.

Reply:
Same issue above not updating